### DISCUSSION TITLE
Why renovate keeps ignoring special packages, e.g. group of `com.android.test` and `com.android.library`

### DISCUSSION THREAD

**Discussion by bingningO** [renovatebot/renovate#32673] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github

### Please tell us more about your question or problem

Hello I want to confirm when a PR will be ignored, we're struggling about it 😅 

Depending on the [Official Doc](https://docs.renovatebot.com/key-concepts/pull-requests/)
>  Renovate creates a new PR when the branch + title "uniqueness" exists again

for most of our dependencies, to make a new version PR be created, while an ignored version(once closed) PR not be created, I guess we must not to delete the branch when closing a PR if we want to ignore it. 

If we delete the branch , depending on the log, will got `branchExists=false`, and any update , no matter if we have ignored it no not, will create a new PR again. 

until here, am I understanding right? 

---

But for a group with `com.android.test` and `com.android.library`, a new version PR will never be created only if we delete the branch.

Our experience is, 
* around Aug 13th, closed a PR to v8.5.2 (not deleting the branch)
* until Nov 8th, found out it never opened a new version PR, tried deleting the branch 
* a new version PR to v8.7.2. is created 

----

I read the log between Aug 13th ~ Nov 8th why a new version PR not be created, 

```
// some process to sync the branch 
DEBUG: syncBranchState() (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: syncBranchState(): Branch cache not found, creating minimal branchState (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: getBranchPr(renovate/android_library_and_test) (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: findPr(renovate/android_library_and_test, undefined, open) (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: findPr(renovate/android_library_and_test, undefined, closed) (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: branchExists=true (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: dependencyDashboardCheck=undefined (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: Check for closed PR because recreating closed PRs is disabled. (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: findPr(renovate/android_library_and_test, chore: Update Android Library and Test to v8.7.2, !open) (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: prAlreadyExisted=false (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: Checking if PR has been edited (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: branch.isModified(): using git to calculate (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: branch.isModified() = true (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
"branchName": "renovate/android_library_and_test",
"unrecognizedAuthors": ["152140799+renovate-agent-for-unext[bot]@users.noreply.github.com"]
DEBUG: findPr(renovate/android_library_and_test, undefined, !open) (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)
DEBUG: Branch has been edited but found no PR - skipping (repository=u-next/UNextAndroid, branch=renovate/android_library_and_test)


// seems the result is pre-edited, so a new PR is not created 
  "branchName": "renovate/android_library_and_test",
  "prNo": null,
  "prTitle": "chore: Update Android Library and Test to v8.7.1",
  "result": "pr-edited",
```

p.s. our configuration is like 
```
  // For PRs
  "prHourlyLimit": 0, // 0 means no limit
  "prConcurrentLimit": 0, // 0 means no limit
  "commitMessagePrefix": "chore: ",
  "pruneBranchAfterAutomerge": true,
  "pruneStaleBranches": true,
  "labels": ["renovate"],
  "reviewers": [
    "team:mad"
  ],
  "automergeStrategy": "merge-commit", // This setting does not work for manual merges.
  "recreateWhen": "never",

  // For Package Updates
  "packageRules": [

     ... 

      {
      "matchPackagePrefixes": [
        "com.android.library",
        "com.android.test"
      ],
      "groupName": "Android Library and Test",
      "groupSlug": "android_library_and_test"
    },

   ... 

  "vulnerabilityAlerts": {
    "enabled": true
  },
  "osvVulnerabilityAlerts": true,
  "major": {
    "minimumReleaseAge": "7 days"
  },
  "minor": {
    "minimumReleaseAge": "3 days"
  },
  "patch": {
    "minimumReleaseAge": "2 days"
  }
}
```

using renovatebot/github-action@v41.0.2 (renovate version 38.52.3)

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Right now the branch is blocked because a branch already exists which has a git author different to the bot. You should delete this branch if you want it unblocked.

**Reply by bingningO** [reply]
thanks for you immediate response ! 

is it because of this line ?  
I didn't get it, as it's the renovate bot we set. 
> "unrecognizedAuthors": ["152140799+renovate-agent-for-unext[bot]@users.noreply.github.com"]

**Reply by rarkins** [reply]
Yes. You must have a configuration error you haven't noticed.

**Reply by bingningO** [reply]
really thanks, let me confirm, mmm happy weekend :)

**Reply by rarkins** [reply]
Thanks, you too. One more bit of info: `152140799+renovate-agent-for-unext[bot]@users.noreply.github.com` is the git author you'll see when GitHub performs a merge on behalf of the bot. Renovate may be expecting to see `renovate-agent-for-unext[bot]@users.noreply.github.com` instead. If you can't solve it, please post more details about your config here later.

**Reply by HonkingGoose** [reply]
Tip: You can use the `description` field to explain what a bit of config does.

See the [Renovate docs, `description`](https://docs.renovatebot.com/configuration-options/#description).

**Reply by bingningO** [reply]
We tried re-set the env variable `RENOVATE_GIT_AUTHOR`, and `unrecognizedAuthors` error not showing any more, I think I can close this discussion. 

really Thanks you guys a lot! 😺
